src/index.js

```diff
@@ -72,6 +84,26 @@ var Index = new Lang.Class({
             doc.add_numeric_value(SEQUENCE_NUMBER_VALUE_NO, metadata['sequenceNumber']);
         }
 
+        const indexingDateString = _indexDateForMetadata(metadata);
+        if (indexingDateString) {
+            /* Parse the date using the native JS Date library, since
+             * GLib.Date.set_parse isn't able to handle JS-style dates. */
+            try {
+                const parsedIndexingDate = new Date(indexingDateString)
+
+                const timestamp = Number([parsedIndexingDate.getUTCFullYear(),
+                                          parsedIndexingDate.getUTCMonth(),
+                                          parsedIndexingDate.getUTCDay(),
+                                          parsedIndexingDate.getUTCHours(),
+                                          parsedIndexingDate.getUTCMinutes(),
+                                          parsedIndexingDate.getUTCSeconds()].map(i => String(i)).join(''));
+                doc.add_numeric_value(PUBLISHED_DATE_VALUE_NO, timestamp);
+            } catch (e) {
+                log('Could not parse ' + indexingDateString + ': ' + String(e) +
```

Comment:
optional: This can be made more readable with string interpolation, and the typecast is not necessary:
```js
log(`Could not parse ${indexingDateString}: ${e}, ${metadata['@id']} will not be date-sortable`);

Reply:
Yeah, good point.